backend/packages/Upgrade/src/api/services/ExperimentAssignmentService.ts

```diff
@@ -1291,9 +1291,10 @@ export class ExperimentAssignmentService {
     }
 
     if (status === MARKED_DECISION_POINT_STATUS.CONDITION_FAILED_TO_APPLY) {
-      const excludeUserDoc: Pick<IndividualExclusion, 'user' | 'experiment' | 'exclusionCode'> = {
+      const excludeUserDoc: Pick<IndividualExclusion, 'user' | 'experiment' | 'groupId' | 'exclusionCode'> = {
         user,
         experiment,
+        groupId: user?.workingGroup[experiment.group],
```

Comment:
`user?.workingGroup?.[experiment.group]` to make sure workingGroup is not null or defined. Same change below